src/main/webapp/app/shared/delete-dialog/delete-button.directive.ts

```diff
@@ -48,14 +52,22 @@ export class DeleteButtonDirective implements OnInit {
             deleteQuestion: this.deleteQuestion,
             deleteConfirmationText: this.deleteConfirmationText,
             additionalChecks: this.additionalChecks,
+            actionType: this.actionType,
+            delete: this.delete,
+            dialogError: this.dialogError,
         };
-        this.deleteDialogService.openDeleteDialog(deleteDialogData).subscribe((additionalChecksValues: { [key: string]: boolean }) => {
-            this.delete.emit(additionalChecksValues);
-        });
+        this.deleteDialogService.openDeleteDialog(deleteDialogData).subscribe(
+            () => {},
```

Comment:
Why subscribe if nothing is done?

Reply:
You are right here in the previous implementation we were emitting delete event here. Now this is not used, so I moved the subscribe to the service itself. There we need to clear the modalRef when the dialog is closed